sched/config/config.go

```diff
@@ -3,445 +3,199 @@ package config
 import (
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
-	"log"
 
-	"github.com/apache/thrift/lib/go/thrift"
-	clusterdef "github.com/scootdev/scoot/cloud/cluster"
-	clusterimpl "github.com/scootdev/scoot/cloud/cluster/memory"
+	"github.com/scootdev/scoot/cloud/cluster"
 	"github.com/scootdev/scoot/saga"
 	"github.com/scootdev/scoot/sched/distributor"
 	"github.com/scootdev/scoot/sched/queue"
-	queueimpl "github.com/scootdev/scoot/sched/queue/memory"
 	"github.com/scootdev/scoot/sched/scheduler"
 	"github.com/scootdev/scoot/sched/worker"
-
-	"github.com/scootdev/scoot/sched/worker/fake"
-	"github.com/scootdev/scoot/sched/worker/rpc"
 	"github.com/scootdev/scoot/scootapi/gen-go/scoot"
 	"github.com/scootdev/scoot/scootapi/server"
-
-	"golang.org/x/net/context"
 )
 
-//
-//TODO: handle and propagate all config values. handle errors. decide extensibility story.
-//
-
-// Return a Config where each field is initialized only if there is no plugin for it.
-func DefaultConfig(cfgFile string, plugins Plugins) (*Config, error) {
-	rawCfg, _ := defaultConfig(cfgFile)
-	realizedCfg := Config{raw: rawCfg, plugins: plugins}
-
-	//TODO: catch errors unmarshaling.
-	if _, ok := plugins[ClusterTypeId]; !ok {
-		var cfg ClusterConfig
-		realizedCfg.Cluster = &cfg
-		json.Unmarshal(rawCfg.Cluster, realizedCfg.Cluster)
-	}
-	if _, ok := plugins[QueueTypeId]; !ok {
-		var cfg QueueConfig
-		realizedCfg.Queue = &cfg
-		json.Unmarshal(rawCfg.Queue, &realizedCfg.Queue)
-	}
-	if _, ok := plugins[SagaCoordTypeId]; !ok {
-		var cfg SagaCoordConfig
-		realizedCfg.SagaCoord = &cfg
-		json.Unmarshal(rawCfg.SagaCoord, &realizedCfg.SagaCoord)
-	}
-	if _, ok := plugins[WorkerTypeId]; !ok {
-		var cfg WorkerConfig
-		realizedCfg.Worker = &cfg
-		json.Unmarshal(rawCfg.Worker, &realizedCfg.Worker)
-	}
-	if _, ok := plugins[PlannerTypeId]; !ok {
-		var cfg PlannerConfig
-		realizedCfg.Planner = &cfg
-		json.Unmarshal(rawCfg.Planner, &realizedCfg.Planner)
-	}
-	if _, ok := plugins[StatsTypeId]; !ok {
-		var cfg StatsConfig
-		realizedCfg.Stats = &cfg
-		json.Unmarshal(rawCfg.Stats, &realizedCfg.Stats)
-	}
-	if _, ok := plugins[HealthTypeId]; !ok {
-		var cfg HealthConfig
-		realizedCfg.Health = &cfg
-		json.Unmarshal(rawCfg.Health, &realizedCfg.Health)
-	}
-
-	return &realizedCfg, nil
+// Config is the top-level configuration for the Scheduler. It defines how to create
+// each of our (configurable) dependencies and then creates the top-level handler.
+type Config struct {
```

Comment:
Health and Stats weren't hooked up only because we have those endpoints running in source, but for whatever reason weren't in the OSS binary. It's just one more thing we can remove from the main file if implemented.